pkg/annotations/middleware.go

```diff
@@ -0,0 +1,111 @@
+package annotations
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+)
+
+type middlewareBuilder func(annotations map[string]string) (middleware *dynamic.Middleware, name string, err error)
+
+// BuildMiddlewares builds middlewares from the given annotations.
+func BuildMiddlewares(annotations map[string]string) (map[string]*dynamic.Middleware, error) {
+	builders := []middlewareBuilder{
+		buildRetryMiddleware,
+		buildRateLimitMiddleware,
+		buildCircuitBreakerMiddleware,
+	}
+
+	middlewares := map[string]*dynamic.Middleware{}
+
+	for _, builder := range builders {
+		middleware, name, err := builder(annotations)
+		if err != nil {
+			return nil, err
+		}
+
+		if middleware != nil {
+			middlewares[name] = middleware
+		}
+	}
+
+	return middlewares, nil
+}
+
+func buildRetryMiddleware(annotations map[string]string) (middleware *dynamic.Middleware, name string, err error) {
+	var retryAttempts int
+
+	retryAttempts, err = GetRetryAttempts(annotations)
+	if err != nil {
+		if err == ErrNotFound {
+			return nil, "", nil
+		}
+
+		return nil, "", fmt.Errorf("unable to build retry middleware: %w", err)
+	}
+
+	name = "retry"
+	middleware = &dynamic.Middleware{
+		Retry: &dynamic.Retry{Attempts: retryAttempts},
+	}
+
+	return middleware, name, nil
```

Comment:
Why are you using named return values if you are not using them?
I think a `return` is enough here (previous return can be simplified too).

Reply:
I followed @Ullaakut suggestion https://github.com/containous/maesh/pull/531#discussion_r419898894

But regarding naked returns, I would prefer not to. It makes the code harder to digest.